fitbit-react/src/Components/VerticalCard/VerticalCard.js

```diff
@@ -1,19 +1,50 @@
 import React from "react";
 import "./verticalcard.css";
-import { MdAddShoppingCart } from "../../Assets/icons";
-import { PrimaryButton } from "../Navbar";
-import SecondaryButton from "../Buttons/SecondaryButton";
-// productTitle={productTitle}
-//             discountedPrice={discountedPrice}
-//             realPrice={realPrice}
-//             discountPercent={discountPercent}
+import {
+  MdAddShoppingCart,
+  PrimaryButton,
+  SecondaryButton,
+  useReducer,
+  RatingBar,
+  useCart,
+} from "./index";
+
 function VerticalCard({
+  productId,
   productTitle,
   discountedPrice,
   realPrice,
   discountPercent,
   productImageUrl,
+  category,
+  inStock,
+  fastDelivery,
+  rating,
 }) {
+  const { cartReducer } = useCart();
+
+  const [cart, dispatch] = useReducer(cartReducer, []);
+  const addToCart = () => {
+    dispatch({
+      type: "ADD_TO_CART",
+      payload: {
+        productId: productId,
```

Comment:
Just a suggestion to reduce the lines of code by sending the product as a whole in and then destructing them in the component.

Reply:
Done